### ISSUE TITLE
Setup wizard to get people to a working instance

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#28630]
Much like the promise of the OSS version of Metabase is to get a instance up and showing real data in under 5 minutes, we should streamline the creating of GTAPs and in general make it easy to set up for customer facing analytics.

Comments from @mazameli 
Quick and dirty designs for this.

[Sketch file here](https://drive.google.com/open?id=1TfTDXX8Cb-MGjq4KnmAhBI_1qJ5hOZEL)

Welcome page
Same welcome page, though I think it would be a good idea to call out and confirm this is the super special paid version of the product somehow. Consider this heading text a placeholder for now:
0 - start
![image](https://user-images.githubusercontent.com/1711649/221378330-32a904e6-0f4c-462f-afc1-a66a65b40f23.png)

Same form for their name etc:
1 - what should we call you
![image](https://user-images.githubusercontent.com/1711649/221378336-b7f33833-fd91-4f97-b7c1-785f6ac3ea0a.png)

Custom branding
I cleaned up the custom branding form @tlrobinson implemented to include a color picker with an input for hex values, and a little div to preview how their logo file will look on the branding color they selected:
2 - custom branding
![image](https://user-images.githubusercontent.com/1711649/221378345-93b93635-7eb5-4fc0-8c70-6cd058f145f8.png)

Same add your data step:
3 - add your data
![image](https://user-images.githubusercontent.com/1711649/221378349-fc7a0c90-aa50-4c20-8362-e2c7ca3dcbfe.png)

Data Sandboxes
The main thing the user can choose to do here is pick the table(s) they want to sandbox. They can skip this step. If they've connected a db with namespaces/schemas, we'll give them a dropdown to pick which schema (like we did for the /explore page). If they didn't add data in the previous step, we'll give them the option to try out sandboxes on the sample dataset.

We'll want to create a saved question for each table they select which they'll use to set up GTAPs. All of these saved questions should go into a single collection called Data Sandboxes. Each saved question's title should be like, Filtering question for {table name}.

The question of when we create those questions and that collection is interesting: seems to me that we should we hold off until they click "Take me to Metabase" so that in case they make edits or alterations on this step we won't have to deal with saved questions that are no longer needed.
4 - sandbox your data
![image](https://user-images.githubusercontent.com/1711649/221378366-bc84e43c-12ea-4fec-9c1c-7e51452d8b8f.png)

4a - sandbox with namespaces
![image](https://user-images.githubusercontent.com/1711649/221378372-7d069f81-4c61-4e8a-8dc5-64c0a5e4b55c.png)

4b - sandbox the sample dataset
![image](https://user-images.githubusercontent.com/1711649/221378376-035a656f-a14e-4c62-811c-c15c00fa4670.png)

SSO setup
This should be relatively simple: there's just a dropdown like with the connect-your-data step, which has a list of all the SSO options we support. In these mocks, I haven't taken the time to mock out each and every form field yet: it seems to me we'll know what these are soon enough and that we should just use the form field components that we're already using for forms in this flow.

5 - set up sso
![image](https://user-images.githubusercontent.com/1711649/221378379-52c3aee1-b2ee-4fb6-84c2-b3f4573619f7.png)

5a - sso selected
![image](https://user-images.githubusercontent.com/1711649/221378383-5334c34a-1fea-4bdb-ac0e-c55e1da56598.png)

Same usage data preferences step:

6 - usage data preferences
![image](https://user-images.githubusercontent.com/1711649/221378384-7a3a853b-3adf-43ca-bd69-36514e758165.png)

You're all set up step
I've removed the newsletter signup from this step because any customer of this product will already be on a special mailing list, and we'll have their email address.

If they've set up data sandboxes, I thought it would be good at the end of the flow here to give them some guidance on what needs to happen next, and a link to the full guide article on how to finish setting up (that page should open in a new window).

7 - setup complete with sandbox info
![image](https://user-images.githubusercontent.com/1711649/221378391-6d0841fd-990b-40d1-84c2-4a8181016d94.png)

If they didn't set up any data sandboxes, then this is super simple:

7a - setup complete didn t do sandboxing
![image](https://user-images.githubusercontent.com/1711649/221378396-c2ba7e10-334b-4857-884a-4957926e03f3.png)

Landing page
The main thing to pain attention to here is that if they set up data sandboxes, the collection containing the filtering questions is present. We're still iterating on the "overworld" screen, so layout here is still in flux.

8 - landing
![image](https://user-images.githubusercontent.com/1711649/221378411-8d63b5e3-40be-40af-8e58-4fd2389ddd51.png)

Comments from @kdoh 
I'm a fairly firm -1 on having all of this be in the initial setup flow. I can maybe see the sandboxes or SSO but I think we oughta keep the initial setup flow as skinny as possible even for the Enterprise and doing branding and making certain other decisions like that all that in the critical path seems like overkill. If we do include branding or other non critical steps they should definitely come after the DB connection so we can at least be doing some work on that in the background).

What're the absolute musts for the enterprise build that need to be set up before they hit the main app? Also how many people on Enterprise Edition will be setting up all the way from the beginning vs upgrading?

Comments from @salsakran 
So the impetus for this was that for a situation where someone is paying for this, they've already invested a fair bit in Metabase, and we don't need to ala cart the entire setup process.

Eg, right now, an instance is fairly lame without setting up email. With Multi-tenant BI considerations, this is exacerbated by how many different places the user needs to go to set this up.

I'm ok with this being a chose your own adventure sorta game, but I very much want the multi-tenant BI setup to be a "15 minutes to glory" sort of deal.

Comments from @kdoh 
"15 minutes to glory" is definitely a good goal. With that as a bit more context I think my main feedback is that a lot of these tasks feel like they could use more space or happen in a series of flows that make doing X task really easy or nice, rather than having to be right after the other. That's why I was curious about what we think absolutely has to happen and in what order for an instance to be "glorious".

Basically my contention is I think we can get people to a well set up instance relatively quickly without it having to happen all in one fell swoop, especially since some of these tasks are non trivial, and don't feel like they fit super naturally inside of the existing structure.

**Comment by brunobergher** [comment]
Handled in many other places, no sense in tracking as an issue here.
